src/compile.c

```diff
@@ -553,6 +564,11 @@ block gen_call(const char* name, block args) {
 
 
 block gen_subexp(block a) {
+  if (block_is_const(a)) {
```

Comment:
You're right.  I'll tweak this to actually check for a `LOADK` instead.  It'll look a little less clean, but it explicitly avoids the case where we accidentally drop a swap.  I don't want this optimization to break things surprisingly in that case.